src/udev/udev-rules.c

```diff
@@ -1830,9 +1830,9 @@ static int udev_rule_apply_token_to_event(
                         bool found = false;
 
                         /* Drop the last line. */
-                        for (char *p = buf + strlen(buf) - 1; p >= buf; p--)
-                                if (strchr(NEWLINE, *p)) {
-                                        *p = '\0';
+                        for (ssize_t i = strlen(buf) - 1; i >= 0; i--)
```

Comment:
the implicit dropping of the sign is a bit sloppy...

Reply:
`size_t` has no sign? What I am more interested in is: can strlen (malloc really) return more data than can fit into `ssize_t`?